dev-tools/xtask/src/common.rs

```diff
@@ -32,3 +32,49 @@ pub fn run_subcmd(mut command: Command) -> Result<()> {
 
     Ok(())
 }
+
+/// Creates and prepares a `std::process::Command` for the `cargo` executable.
+pub(crate) fn cargo_command() -> Command {
+    let cargo =
+        std::env::var("CARGO").unwrap_or_else(|_| String::from("cargo"));
+    let mut command = Command::new(&cargo);
+
+    for (key, _) in std::env::vars_os() {
+        let Some(key) = key.to_str() else { continue };
+        if SANITIZED_ENV_VARS.matches(key) {
+            command.env_remove(key);
+        }
+    }
+
+    command
+}
+
+#[derive(Debug)]
+struct SanitizedEnvVars {
+    // At the moment we only ban some prefixes, but we may also want to ban env
+    // vars by exact name in the future.
+    prefixes: Vec<&'static str>,
```

Comment:
This doesn't need to be a Vec, right?

```suggestion
    prefixes: &'static [&'static str],
```

Then `SanitizedEnvVars::new` could be a `const fn` and we don't need a `LazyLock`.

Reply:
True -- I originally had it as a `BTreeSet` which is why it's this way, haha.

Reply:
Fixing.